src/kvstore/comm.h

```diff
@@ -170,34 +170,36 @@ class CommCPU : public Comm {
     }
   }
 
-  // TODO(haibin) support broadcast row_sparse on GPU
   void BroadcastRowSparse(int key, const NDArray& src,
                           const std::vector<std::pair<NDArray*, NDArray>>& dst,
                           const bool use_copy,
                           const int priority) override {
     using namespace mshadow;
-    auto size = dst.size();
-    for (size_t i = 0; i < size; i++) {
-      auto out = dst[i].first;
-      auto row_id = dst[i].second;
+    for (size_t i = 0; i < dst.size(); ++i) {
```

Comment:
I was wondering how much time could be saved using cached size and how large the size should be in order to see the difference? What is the compiler version?